app/models/link.rb

```diff
@@ -1,4 +1,18 @@
 class Link < ApplicationRecord
   has_and_belongs_to_many :jobs
+  has_many :checks
+
   validates_presence_of :uri
+
+  def existing_check
+    checks
+      .where(ended_at.gt(24.hours.ago))
+      .first
+  end
```

Comment:
Also in a different area of concern, I wonder if we want ended_at to be `completed_at` as I think that's what we have on a Job (or batch)